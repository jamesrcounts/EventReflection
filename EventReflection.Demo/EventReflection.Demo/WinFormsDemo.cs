namespace EventReflection.Demo
{
    using System;

    using ApprovalTests.WinForms;

    using Microsoft.VisualStudio.TestTools.UnitTesting;

    [TestClass]
    public class WinFormsDemo
    {
        [TestMethod]
        public void VerifyDemoFormView()
        {
            WinFormsApprovals.Verify(new DemoForm());
        }

        [TestMethod]
        public void GetEventTypeForDemoForm()
        {
            ApprovalTests.Approvals.VerifyAll(
                ReflectionUtility.GetEventTypes(new DemoForm()), string.Empty);
        }

        [TestMethod]
        public void GetEventsForDemoFormEventHandlers()
        {
            ApprovalTests.Approvals.VerifyAll(
                new DemoForm().GetEventsForTypes(typeof(EventHandler)),
                string.Empty);
        }

        [TestMethod]
        public void VerifyDemoFormEvents()
        {
            EventUtility.VerifyEventCallbacks(new DemoForm());
        }
    }
}