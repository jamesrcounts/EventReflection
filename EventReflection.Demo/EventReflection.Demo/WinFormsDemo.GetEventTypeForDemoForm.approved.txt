[0] = System.EventHandler
[1] = System.ComponentModel.CancelEventHandler
[2] = System.Windows.Forms.FormClosingEventHandler
[3] = System.Windows.Forms.FormClosedEventHandler
[4] = System.Windows.Forms.InputLanguageChangedEventHandler
[5] = System.Windows.Forms.InputLanguageChangingEventHandler
[6] = System.Windows.Forms.ScrollEventHandler
[7] = System.Windows.Forms.ControlEventHandler
[8] = System.Windows.Forms.DragEventHandler
[9] = System.Windows.Forms.GiveFeedbackEventHandler
[10] = System.Windows.Forms.HelpEventHandler
[11] = System.Windows.Forms.InvalidateEventHandler
[12] = System.Windows.Forms.PaintEventHandler
[13] = System.Windows.Forms.QueryContinueDragEventHandler
[14] = System.Windows.Forms.QueryAccessibilityHelpEventHandler
[15] = System.Windows.Forms.KeyEventHandler
[16] = System.Windows.Forms.KeyPressEventHandler
[17] = System.Windows.Forms.LayoutEventHandler
[18] = System.Windows.Forms.MouseEventHandler
[19] = System.Windows.Forms.PreviewKeyDownEventHandler
[20] = System.Windows.Forms.UICuesEventHandler
